doc/modules/linear_model.rst

```diff
@@ -842,6 +848,98 @@ Ridge Regression`_, see the example below.
 
     .. [4] Tristan Fletcher: `Relevance Vector Machines Explained <https://citeseerx.ist.psu.edu/doc_view/pid/3dc9d625404fdfef6eaccc3babddefe4c176abd4>`_
 
+.. _linear_regression_intercept:
+
+Note on the handling of the intercept in linear regression models
+=================================================================
+
+All least-squares linear regression estimators in scikit-learn solve a centered
+version of the problem without intercept and then compute the intercept a
+posteriori from that solution.
+
+This ensures that the ridge solution converges to the minimal norm OLS solution
+when `alpha` tends to zero in the underdeterimined case (more features than
+samples).
+
+The following explains why this approach is valid, both in the penalized and
+unpenalized underdetermined cases.
+
+Intercept of the penalized least-squares solution
+-------------------------------------------------
+
+Recall the definition of the ridge estimator:
+
+.. math::  \min_{w, w_0} { ||X w + w_0 1_s - y||_2 ^ 2 + \alpha ||w||_2 ^ 2}
+
+Here :math:`1_s` is a vector of size `n_samples` where all entries are 1.
+:math:`w_0` is a scalar parameter and is not regularized.
+
+Let's note :math:`\bar{X}` the column vector whose entries the means of the
+columns of :math:`X` and :math:`\bar{y}` the mean scalar value of :math:`y`,
+:math:`X_c` the centered version of :math:`X`, :math:`y_c` the centered version
+of :math:`y` such that:
+
+.. math::  X = X_c + 1_s \bar{X}^{T}  \text{ and } y = y_c + \bar{y} 1_s
+
+Optimizing the original ridge objective with intercept is equivalent to solving
+the following ridge problem on the centered dataset with no intercept:
+
+.. math:: \hat{w} = \min_{w} { ||X_c w - y_c||_2 ^ 2 + \alpha ||w||_2 ^ 2}
+
+and subsequently define the intercept as :math:`\hat{w_0} = \bar{y} -
+\bar{X}^{T} \hat{w}`.
+
+To see this, let's rewrite the data-fit term of the original ridge problem as
+follows (after rearranging the terms and factorizing by :math:`1_s`):
```

Comment:
As @agramfort mentioned, just taking the derivative of $|| X w + w_0 - y||_2^2$ w.r.t. $w_0$ might be an easier derivation.